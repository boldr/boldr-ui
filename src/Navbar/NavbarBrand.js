/* @flow */
import React from 'react';
import styled from 'styled-components';
import cxN from 'classnames';

export type Props = {
  isFixed: boolean,
};

const NavbarBrand = props => {
  return (
    <li className="boldrui-navbar-header">
      <a className="boldrui-navbar-brand">Boldr</a>
      <label
        id="boldrui-navbar-hamburger"
        className="boldrui-navbar-hamburger boldrui-navbar-hamburger-doublespin"
        htmlFor="boldrui-navbar-checkbox"
      >
        <span />
      </label>
    </li>
  );
};

export default NavbarBrand;
