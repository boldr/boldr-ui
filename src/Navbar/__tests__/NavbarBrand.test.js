import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavbarBrand } from '../NavbarBrand';

describe('<NavbarBrand />', () => {
  it('should render a div with .boldr-navbar__brand', () => {
    const wrapper = shallow(<NavbarBrand>Test</NavbarBrand>);
    expect(wrapper.contains(<div className="boldr-navbar__brand">Test</div>)).toBe(true);
  });

  it('should render a p with .navbar-brand', () => {
    const wrapper = shallow(<NavbarBrand tag="p">Test</NavbarBrand>);
    expect(wrapper.contains(<p className="boldr-navbar__brand">Test</p>)).toBe(true);
  });

  it('should render a div with .navbar-brand and custom classNames', () => {
    const wrapper = shallow(
      <NavbarBrand className="custom">
        <span>Test</span>
      </NavbarBrand>,
    );
    expect(wrapper.hasClass('boldr-navbar__brand')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavbarBrand>Test</NavbarBrand>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
