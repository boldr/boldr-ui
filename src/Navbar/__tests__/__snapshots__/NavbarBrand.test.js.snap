// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarBrand /> should match the snapshot 1`] = `
<div
  className="boldrui-navbar__brand"
>
  Test
</div>
`;
